sycl/plugins/unified_runtime/ur/usm_allocator.cpp

```diff
@@ -542,13 +269,17 @@ class USMAllocContext::USMAllocImpl {
   // Store as unique_ptrs since Bucket is not Movable(because of std::mutex)
   std::vector<std::unique_ptr<Bucket>> Buckets;
 
+  // Configuration for this instance
+  USMAllocatorParameters params;
+
 public:
-  USMAllocImpl(std::unique_ptr<SystemMemory> SystemMemHandle)
-      : MemHandle{std::move(SystemMemHandle)} {
+  USMAllocImpl(std::unique_ptr<SystemMemory> SystemMemHandle,
+               USMAllocatorParameters params)
+      : MemHandle{std::move(SystemMemHandle)}, params(params) {
 
     // Generate buckets sized such as: 64, 96, 128, 192, ..., CutOff.
     // Powers of 2 and the value halfway between the powers of 2.
-    auto Size1 = USMSettings.MinBucketSize[MemHandle->getMemType()];
+    auto Size1 = params.MinBucketSize;
```

Comment:
would this affect performance during this period of transition, if MinBucketSize may be different depending on the memory type? See here
https://github.com/intel/llvm/blob/8759d0728929c4e8aae88fddb06206aad5719d2b/sycl/plugins/unified_runtime/ur/usm_allocator.cpp#L101

Reply:
No, in fact, my goal was to keep the original behavior intact. This assignment is just in a different place: https://github.com/intel/llvm/blob/51609155370086670b74ba99f3724d3d8d340852/sycl/plugins/unified_runtime/ur/usm_allocator_config.cpp

In the original implementation, those variables (MinBucketSize, Capacity, itp.) were stored in a single global structure and indexed using `[getMemType()]`. I have moved those variables to a separate class: `USMAllocatorParameters` and I'm now passing instance of this class to the USMAllocContext as you can see here: https://github.com/intel/llvm/blob/51609155370086670b74ba99f3724d3d8d340852/sycl/plugins/level_zero/pi_level_zero.cpp#L844

Basically, instead of having Structure of Arrays:

```
class SetLimits() {
   ...
   size_t MinBucketSize[MemType::All];
};
```

we now have Array of Structs:

```
class USMAllocatorParameters {
   ...
  size_t MinBucketSize;
} configs[MemType::All];
```

Reply:
The goal of this is to split the pool management logic from reading/setting the configuration since this might differ for other adapters.

Reply:
thanks @igchor.